scripts/dupe_finder.py

```diff
@@ -0,0 +1,63 @@
+import json
+import os
+import hashlib
+
+
+def getFileHash(filename, alg, chunksize=131072):
+    if (alg == 'sha256'):
+        h = hashlib.sha256()
+    elif (alg == 'sha1'):
+        h = hashlib.sha1()
+    elif (alg == 'md5'):
+        h = hashlib.md5()
+
+    with open(filename, 'rb', buffering=0) as f:
+        for b in iter(lambda: f.read(chunksize), b''):
+            h.update(b)
+    return h.hexdigest()
+
+
+'''
+    check if rom is present in the list of files in json
+    return: -1 if no rom is listed, rom's filename if found
+'''
+
+
+def lookForRom(files):
+    for f in range(0, len(files)):
+        ext = files[f]['filename'].split('.')[-1]
+        if ext.lower() in ['gb', 'gbc', 'cgb', 'gba', 'agb', 'sgb']:
+            return files[f]['filename']
+
+    return -1
+
+
+# dictionary to keep track of entries
+# keys -> hash
+# values -> slugs
+# if a homebrew is duped, then it should have more than one slugs
+d = dict()
+
+for folder in os.listdir('../entries'):
+    with open('../entries/'+folder+'/game.json') as f:
+        data = json.load(f)
+
+    gamePath = '../entries/'+folder+'/'
+
+    rom_name = lookForRom(data['files'])
+
+    if(rom_name != -1):
```

Comment:
What is happening here? Can you describe the logic?

Reply:
if rom has been found (see my previous comment), get its hash (we cannot assume it is already saved because it could be missing).

a dictionary called d has been created since dictionaries are really useful to detect unique things.
We need to know what Roms are the same: one way to do it is to check their hashes.
If hash is not in d, let's add the hash as a key and a list of a value.
You will then append the slug to the list with hash as a key.

At the end, if a list contains more than one slug, it means that those slugs are the same, because I"ve met twice (or more times) the same rom, because every time I've encountered a rom I saved its hash.

If it is still not clear, I can provide an example or explain more

Reply:
Works for me! Can you add it directly in the code as comment?